core/src/main/java/io/questdb/cairo/CairoSecurityContext.java

```diff
@@ -24,5 +24,15 @@
 
 package io.questdb.cairo;
 
+import io.questdb.cairo.security.CairoSecurityException;
+
 public interface CairoSecurityContext {
+
+    public static void checkWritePermission(CairoSecurityContext securityContext) {
+        if (null == securityContext || !securityContext.canWrite()) {
+            throw CairoSecurityException.instance().put("Write permission denied");
```

Comment:
Generally speaking nothing in the codebase handles exceptions via Throwable. The rule of thumb when new exception is introduced a special consideration needs to be paid to how this exception is received by API users. This special consideration is missing from the PR.

Easier way to deal with this is to reuse CairoException this is also a runtime exception and it is handled (and handling is tested)

Reply:
Yes, I can see how this breaks the GUI, I didn't notice that. I had originally used a CairoException but the problem with that is that SqlCompiler.alterTable catches CairoException and assumes its due to a failed table lock and then throws a SqlException which is wrong due to the incorrect assumption.

In my view I should be using a CairoException, but the behaviour of SqlCompiler.alterTable is wrong, we shouldn't be making assumptions on the reasons for exceptions